google-cloud-firestore/src/main/java/com/google/cloud/firestore/TransactionRunner.java

```diff
@@ -146,24 +146,29 @@ public void run() {
    */
   private SettableApiFuture<T> invokeUserCallback() {
     final SettableApiFuture<T> callbackResult = SettableApiFuture.create();
+
     userCallbackExecutor.execute(
         new Runnable() {
           @Override
           public void run() {
-            ApiFutures.addCallback(
-                userCallback.updateCallback(transaction),
-                new ApiFutureCallback<T>() {
-                  @Override
-                  public void onFailure(Throwable t) {
-                    callbackResult.setException(t);
-                  }
+            try {
+              ApiFutures.addCallback(
+                  userCallback.updateCallback(transaction),
```

Comment:
I think this code would be easier to understand if you did something like this:

```
try {
const result = userCallback.updateCallback(transaction)
ApiFutures.addCallback(...)
} catch (Exception e) {
   callbackResult.setException(e);
}
```

It's very hard to tell what the try/catch block addresses. The other alternative is to wrap the callback in a try/catch before passing it here like in TransactionAsyncFunction.

Reply:
Done, moved it out into a separate variable.